### DISCUSSION TITLE
[Toast] Is it possible (or discouraged) to use nested ToastProviders?

### DISCUSSION THREAD

**Discussion by grelas** [radix-ui/primitives#3138] [Help]
Similar to https://github.com/radix-ui/primitives/discussions/1282, I wrap my application in the `ToastProvider` to manage global toasts. However, we also want the ability to show a Toast within a dialog modal alongside the global toast viewport. When I attempted to use nested ToastProviders, it does end up working, but wondering if this should be avoided - something like:
```
<ToastProvider>
   <App>
      <GlobalToasts />
        <ToastProvider>
           <ToastViewport>
      <Page>
          <Modal>
            <ModalToasts /> <-- these toasts would be positioned relative to the Modal
              <ToastProvider>
                <ToastViewport>
```

If not, I'd suggest alongside "Features" section, there could be a "Limitations" section outlining this.

**Comment by wbobeirne** [comment]
I don't see it explicitly called out in the React docs on contexts, but it's totally safe to nest multiple contexts. Your components will just use the most closest context in their ancestry.
